### DISCUSSION TITLE
Notify zone to own Theme

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#8772] [Q&A]
Please help me to add Notify zone to my own user theme.
Theme build using TheTheme. as example.
I have 
`  @await RenderSectionAsync("Messages", required: false)`
in layout cshtml.
From my module (Razor page) 
`_notifier.Error(T["Please check and try again."]);`

**Comment by vitalybrandes** [comment]
anyone?

**Reply by luis3199correia** [reply]
Hi,
I'm using the Orchard Notify adding this to my controller
`public TokensController(INotifier notifier, IHtmlLocalizer<TokensController> localizer)
{
	this.Notifier = notifier;
	this.Localizer = localizer;
}`
`public TokensController(INotifier notifier, IHtmlLocalizer<TokensController> localizer)
{
	if(tokenIsRemoved)
        {
	this.Notifier.Success(this.Localizer["Your token has been removed."]);
        }
}`

**Reply by vitalybrandes** [reply]
@luis3199correia 
I am using it form the Razor page.
Did you try to do the same from razor page?

**Reply by luis3199correia** [reply]
No, I've created a theme to add new functionalities to Orchard, like 'tokens' CRUD pages, and added that pages to orchard AdminMenu. Therefore, adding the previous code to the controller, it is enough for the orchard to add the notification.
Hope this helps

**Reply by vitalybrandes** [reply]
thank you,
but look like this not work for razor pages

**Comment by deanmarcussen** [comment]
I'm not sure the Notify Filter will work from a razor page. It might only work from controllers at the moment

It's not an `IAsyncPageFilter`, only an `IAsyncResultFilter`, so might have to be upgraded to support razor pages. (not tested, but I think so)

Please open an issue, or pr

**Reply by vitalybrandes** [reply]
I will, thank you!

**Reply by vitalybrandes** [reply]
#8795
